python/pycylon/data/table.pyx

```diff
@@ -213,14 +248,112 @@ cdef class Table:
+                self.jcPtr = new CJoinConfig(CJoinType.CINNER, left_column_index,
+                                             right_column_index, CJoinAlgorithm.CSORT)
+            elif join_type == PJoinType.LEFT.value:
+                self.jcPtr = new CJoinConfig(CJoinType.CLEFT, left_column_index,
+                                             right_column_index, CJoinAlgorithm.CSORT)
+            elif join_type == PJoinType.RIGHT.value:
+                self.jcPtr = new CJoinConfig(CJoinType.CRIGHT, left_column_index,
+                                             right_column_index, CJoinAlgorithm.CSORT)
+            elif join_type == PJoinType.OUTER.value:
+                self.jcPtr = new CJoinConfig(CJoinType.COUTER, left_column_index,
+                                             right_column_index, CJoinAlgorithm.CSORT)
+            else:
+                raise ValueError("Unsupported Join Type {}".format(join_type))
+        else:
+            if join_type == PJoinType.INNER.value:
+                self.jcPtr = new CJoinConfig(CJoinType.CINNER, left_column_index,
+                                             right_column_index)
+            elif join_type == PJoinType.LEFT.value:
+                self.jcPtr = new CJoinConfig(CJoinType.CLEFT, left_column_index, right_column_index)
+            elif join_type == PJoinType.RIGHT.value:
+                self.jcPtr = new CJoinConfig(CJoinType.CRIGHT, left_column_index,
+                                             right_column_index)
+            elif join_type == PJoinType.OUTER.value:
+                self.jcPtr = new CJoinConfig(CJoinType.COUTER, left_column_index,
+                                             right_column_index)
+            else:
+                raise ValueError("Unsupported Join Type {}".format(join_type))
+
+    cdef shared_ptr[CTable] init_join_ra_params(self, table, join_type, algorithm, kwargs):
+        left_cols, right_cols = self._get_join_column_indices(table=table, **kwargs)
+
+        # Cylon only supports join by one column and retrieve first left and right column when
+        # resolving join configs
+        self.__get_join_config(join_type=join_type, join_algorithm=algorithm,
+                               left_column_index=left_cols[0],
+                               right_column_index=right_cols[0])
+        cdef shared_ptr[CTable] right = pycylon_unwrap_table(table)
+        return right
+
+    cdef _get_join_ra_response(self, op_name, shared_ptr[CTable] output, CStatus status):
+        if status.is_ok():
+            return pycylon_wrap_table(output)
+        else:
+            raise ValueError(f"{op_name} operation failed!")
+
+    cdef _get_ra_response(self, table, ra_op_name):
+        cdef shared_ptr[CTable] output
+        cdef shared_ptr[CTable] right = pycylon_unwrap_table(table)
+        cdef CStatus status
+        # TODO: add callable for Cython functions via FPointers
+
+        if ra_op_name == 'union':
```

Comment:
As I said earlier, I would still prefer a switch case with an enum here. or better yet, use a function pointer (I am not sure how that could be done in cython though). Latter would eliminate the switch case, and it would be elegant IMO.